Dockerfile

```
@@ -1,14 +1,21 @@
-FROM node:16
+# Build Stage
+FROM node:18-alpine AS build-stage
 WORKDIR /usr/src/cross-seed
-RUN npm install -g npm@9
 COPY package*.json ./
-RUN npm ci
-ENV CONFIG_DIR=/config
-ENV DOCKER_ENV=true
+RUN npm install -g npm@9 \
+    && npm ci
 COPY tsconfig.json tsconfig.json
 COPY src src
-RUN npm run build
+RUN npm run build \
+    && rm -rf src tsconfig.json
+
+# Production Stage
+FROM node:18-alpine
+WORKDIR /usr/src/cross-seed
+COPY --from=build-stage /usr/src/cross-seed .
 RUN npm link
+ENV CONFIG_DIR=/config
+ENV DOCKER_ENV=true
 EXPOSE 2468
 WORKDIR /config
-ENTRYPOINT ["cross-seed"]
+ENTRYPOINT ["cross-seed"]
```

nit: end all files with LF

```suggestion
ENTRYPOINT ["cross-seed"]

```